### ISSUE TITLE
libcurl memory leak

### ISSUE THREAD

**Issue by okoshovetc** [tarantool/tarantool#4389]
Tarantool version:
```
$ tarantool --version
Tarantool 1.10.3-104-g1aa8b92
Target: Linux-x86_64-RelWithDebInfo
Build options: cmake . -DCMAKE_INSTALL_PREFIX=/usr -DENABLE_BACKTRACE=ON
Compiler: /opt/rh/devtoolset-6/root/usr/bin/cc /opt/rh/devtoolset-6/root/usr/bin/c++
C_FLAGS:-O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches   -m64 -mtune=generic  -fexceptions -funwind-tables -fno-omit-frame-pointer -fno-stack-protector -fno-common -fopenmp -msse2 -std=c11 -Wall -Wextra -Wno-strict-aliasing -Wno-char-subscripts -Wno-format-truncation -fno-gnu89-inline -Wno-cast-function-type
CXX_FLAGS:-O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches   -m64 -mtune=generic  -fexceptions -funwind-tables -fno-omit-frame-pointer -fno-stack-protector -fno-common -fopenmp -msse2 -std=c++11 -Wall -Wextra -Wno-strict-aliasing -Wno-char-subscripts -Wno-format-truncation -Wno-invalid-offsetof -Wno-cast-function-type
```

OS version: 
```
$ hostnamectl | grep System
  Operating System: CentOS Linux 7 (Core)
```

Bug description:
Some versions of libcurl provide a memleak on https requests. This is not a bug, more a reproducer for ticket https://github.com/tarantool/tarantool/issues/4318

Steps to reproduce:
```
$ cat init.lua
-- $ rpm -qa | grep libcurl
-- libcurl-7.29.0-42.el7_4.1.x86_64

-- $ tarantool --version
-- Tarantool 1.10.3-104-g1aa8b92
-- Target: Linux-x86_64-RelWithDebInfo
-- Build options: cmake . -DCMAKE_INSTALL_PREFIX=/usr -DENABLE_BACKTRACE=ON
-- Compiler: /opt/rh/devtoolset-6/root/usr/bin/cc /opt/rh/devtoolset-6/root/usr/bin/c++
-- C_FLAGS:-O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches   -m64 -mtune=generic  -fexceptions -funwind-tables -fno-omit-frame-pointer -fno-stack-protector -fno-common -fopenmp -msse2 -std=c11 -Wall -Wextra -Wno-strict-aliasing -Wno-char-subscripts -Wno-format-truncation -fno-gnu89-inline -Wno-cast-function-type
-- CXX_FLAGS:-O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches   -m64 -mtune=generic  -fexceptions -funwind-tables -fno-omit-frame-pointer -fno-stack-protector -fno-common -fopenmp -msse2 -std=c++11 -Wall -Wextra -Wno-strict-aliasing -Wno-char-subscripts -Wno-format-truncation -Wno-invalid-offsetof -Wno-cast-function-type

box.cfg{
	listen = '127.0.0.1:3333';
}

box.schema.user.grant('guest', 'read,write,execute,create,drop', 'universe')

_G.STOP_MEMTEST = false

local http_request = require 'http.client'.request
local fiber        = require 'fiber'

for i = 1, 10 do
	fiber.create(function(i)
		fiber.self():name(("%s:memleak"):format(i))
		while true do
			if not _G.STOP_MEMTEST then
				http_request('GET', 'https://ya.ru')
			else
				fiber.sleep(10)
			end
		end
	end, i)
end

```

```
$ tarantool init.lua
```

BTW upgrading to `libcurl-7.29.0-51.el7.x86_64` seems to slow down the leak but not stopping it completely

**Comment by ImeevMA** [comment]
After investigation, it turned out that curl in Centos 7 was built with the ```--without-ssl``` and ```--with-nss``` options, and this led to a memory leak. This problem was resolved by adding curl statically with the ```--with-ssl``` and ```--without-nss``` options.

**Comment by Totktonada** [comment]
So I think we found the cause of the problem and can be sure that all current branches (1.10, 2.1, 2.2, master) are not affected by this issue. So it is closed in #4318.
